### PR TITLE
Added ModelsName Setting parameter and update CSharp to use this in place of the hard coded "Models"

### PR DESCRIPTION
**Pull request by John-Hart**
This Fixes issue #470 for CSharp. 
CSharp will now use the ModelsName for models output folder and models namespace

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@John-Hart__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

File: src/core/AutoRest.Core/Settings.cs [hunk 1]
```diff
@@ -81,6 +82,13 @@ public Settings()
         public string Input { get; set; }
 
         /// <summary>
+        /// Gets or sets a name for the generated client models Namespace and Models output folder
+        /// </summary>
+        [SettingsInfo("Name to use for the generated client models namespace and folder name. Not supported by all code generators.")]
```
Comment (fearthecowboy) [inline src/core/AutoRest.Core/Settings.cs]: I'm wondering if we _want_ to force the `models` folder as a sub-folder/sub-namespace of the parent (ie, by doing it as you've done here, declared as just the last `.part` ), or if we should allow them to specify the whole namespace/foldername. (ie, `-ModelsName Foo.Bar.Bin.Baz` ) 

That would require the code generator to calculate the folder name a bit more carefully  (ie, generated relative between the `client` namespace/path and the `models` namespace/path.

And does this let us choose the same namespace for the client as the model itself? Is it possible that someone **would** want that?

(food for thought...)

File: src/core/AutoRest.Core/Settings.cs [hunk 2]
```diff
@@ -81,6 +82,13 @@ public Settings()
...
+        [SettingsAlias("mname")]
+        public string ModelsName { get; set; }
```
Comment (fearthecowboy) [inline src/core/AutoRest.Core/Settings.cs]: And if this was the whole namespace (as opposed to just the suffix) then you wouldn't have to do string smashing down below...

File: src/generator/AutoRest.CSharp/Templates/EnumTemplate.cshtml [hunk 3]
```diff
@@ -2,7 +2,7 @@
 @inherits AutoRest.Core.Template<AutoRest.CSharp.TemplateModels.EnumTemplateModel>
 @Header("// ")
 @EmptyLine
-namespace @(Settings.Namespace).Models
+namespace @(Settings.Namespace).@(Settings.ModelsName)
```
Comment (fearthecowboy) [inline src/generator/AutoRest.CSharp/Templates/EnumTemplate.cshtml]: (the string smashing I was referring to)

File: src/generator/AutoRest.CSharp.Unit.Tests/Bug470.cs [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace AutoRest.CSharp.Unit.Tests
+{
+    public class Bug470 : BugTest
```
Comment (fearthecowboy) [inline src/generator/AutoRest.CSharp.Unit.Tests/Bug470.cs]: (ignore this comment, just checking for codeflow vs github differences)

**Comment by fearthecowboy** [comment]
LGTM :+1:

**Pull request merged**